super_editor/example/lib/demos/custom_elements/task_item.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:async';
+
+import 'package:example/demos/custom_elements/task.dart';
+import 'package:flutter/material.dart';
+import 'package:super_editor/super_editor.dart';
+
+/// A [DocumentNode] representing a [Task] in a [Document].
```

Comment:
```suggestion
/// A [TextNode] representing a [Task] in a [Document].
```

I would suggest to state that this is a TextNode rather than DocumentNode as it is extending itself from it. People can direct themselves to DocumentNode from TextNode later on.